### ISSUE TITLE
BUG: Local repo (connected to an origin, although I would like not to have to) - Cannot get remote.origin.url.

### ISSUE THREAD

**Issue by hidegh** [codersrank-org/repo_info_extractor#170]
```
PS C:\My work\! Codersrank.io> .\test.ps1
repo : c:\$_Work\!_codersrank\FIS-X
out  : ..\FIS-X.2021-04-08.repo_data.json
Initializing repository
Cannot get remote.origin.url. Use directory path to get repo name.
Analysing commits
fatal error: all goroutines are asleep - deadlock!

goroutine 1 [select]:
github.com/codersrank-org/repo_info_extractor/extractor.(*RepoExtractor).getCommits.func1(0xc00053c060, 0xc00011dae8, 0x3e8, 0xc00053c000, 0xc00011db18, 0xc00053c0c0, 0xc00011dad8)
        C:/z Archive/Work/! Codersrank.io/repo_info_extractor/extractor/extractor.go:289 +0xb5
github.com/codersrank-org/repo_info_extractor/extractor.(*RepoExtractor).getCommits(0xc0002ec2d0, 0x0, 0x0, 0x197c760, 0x1, 0x12)
        C:/z Archive/Work/! Codersrank.io/repo_info_extractor/extractor/extractor.go:305 +0x205
github.com/codersrank-org/repo_info_extractor/extractor.(*RepoExtractor).analyseCommits(0xc0002ec2d0, 0x0, 0x0)
        C:/z Archive/Work/! Codersrank.io/repo_info_extractor/extractor/extractor.go:200 +0xdb
github.com/codersrank-org/repo_info_extractor/extractor.(*RepoExtractor).Extract(0xc0002ec2d0, 0xc0002ec2d0, 0x0)
        C:/z Archive/Work/! Codersrank.io/repo_info_extractor/extractor/extractor.go:62 +0x8b
main.main()
        C:/z Archive/Work/! Codersrank.io/repo_info_extractor/main.go:48 +0x53c
exit status 2
```

using latest release source codes (same error on the windows exe) test:
```
go test ./...
?       github.com/codersrank-org/repo_info_extractor   [no test files]
?       github.com/codersrank-org/repo_info_extractor/commit    [no test files]
?       github.com/codersrank-org/repo_info_extractor/emailsimilarity   [no test files]
ok      github.com/codersrank-org/repo_info_extractor/extractor 0.133s
?       github.com/codersrank-org/repo_info_extractor/languagedetection [no test files]
?       github.com/codersrank-org/repo_info_extractor/librarydetection  [no test files]
ok      github.com/codersrank-org/repo_info_extractor/librarydetection/languages        0.103s
?       github.com/codersrank-org/repo_info_extractor/obfuscation       [no test files]
```

**Comment by peti2001** [comment]
Hi @hidegh,

Like this, it will be difficult the debug. Is it happening with other repositories or only with that one?

**Comment by hidegh** [comment]
Hi @peti2001 - it's happening with others too, tried local github and bitbucket.

**Questions:**
1. does the tool need active connection to the repo or is it working on local git repo folder only (as the old version - which is easier, no need to worry about the connection)
2. can the special chars ! or $ be a reason for this
3. is there a way to get more **verbose output (debug)?** I have GO installed (as unfortunately you moved from PY to GO)...so I could provide you with more details (IMO the exception should have been enough, if not, then there's an issue in the code :)

**Comment by peti2001** [comment]
1. No it doesn't require connection, it read everything from the local git repo.
3. I pushed some changes to improve the error logging. Can you please run the code again.

**Comment by hidegh** [comment]
Köszi @peti2001,

👍 It worked (extended traces helped), so here are my **notes**:

1. I did not get that error as previously...
2. ...got GIT was not found, so had to use **-git_path**
3. ❗ Unlinke the old python version, there was no real feedback while **analysing libraries** and thus for minutes (i7, 32GB RAM Win 10) I wasn't sure if the app stalled or not
4. ❗ At the end I got: "Uploading result to CodersRank" - but what if I don't wan't to immediately upload (old code had a -do_not_upload switch) - also to note a 🐛: the new file was not uploaded to codersrank, as the old file with the same name and old timestamp was kept there.
5. 🆕 It would be cool to have a way to alter the display name of the repo (make it nicer) what about a **-display_name_override** option?
6. ℹ️ More details on exception and errors are always useful.

Plz. @peti2001 let me know **if/when Nr. 4 and Nr. 5 will be added.** If so, then this tool might be even better than the previous one. Also would consider to add some progress while analysing, see Nr 3.

For others, who plant to use multiple project extraction, here's a simple powershell scritpt I use:
```
$gitPath = "c:\Program Files\Git\bin\git.exe"
$timestamp = Get-Date -Format "yyyy-MM-dd"
$emails = "comma@delimited.emails"

$repos = @(
"c:\Temp\FinIS"
)

$winExe = $False

if (!$winExe)
{
  cd repo_info_extractor
}

$repos | ForEach-Object {
  $repo = "$_"
  $name = Split-Path $repo -leaf
  $out = "$name.$timestamp.repo_data.json"
  if (!$winExe)
  {
    $out = "..\" + $out
  }
  Write-Host "repo : $repo"
  Write-Host "out  : $out"
  if (! $winExe)
  {
    go run . --repo_path "$repo" --output_path "$out" --emails "$emails" -git_path "$gitPath"
  }
  else
  {
    & ".\repo_info_extractor_windows.exe" -repo_path "$repo" -output_path "$out" -emails "$emails" -git_path "$gitPath"
  }
}

if (!$winExe)
{
  cd ..
}
```
